Simplenote/Classes/SPNoteListViewController.m

```diff
@@ -431,6 +444,9 @@ - (void)openNote:(Note *)note ignoringSearchQuery:(BOOL)ignoringSearchQuery anim
 {
     [SPTracker trackListNoteOpened];
 
+    self.selectedNote = note;
```

Comment:
Q: do you think we could, maybe, avoid keeping a reference here?

Maybe something in the lines of...

```
let editor = navigationController.viewControllers.first { $0 is NoteEditorViewController }
return editor?.note
```

Maybe not worthy? (just trying to avoid the need to "maintain" yet another property)

Reply:
Yeah, this part is ugly. But right now I don't have any better ideas :-(

At first I also thought about checking which note is now in the editor and use this information to restore selected note when we come back to the list. If I do this in `viewWillAppear` method of the list view controller, there is no editor because it was already removed from the navigation controller stack.